src/ashell/comms-uart.c

```diff
@@ -313,16 +313,20 @@ void comms_write_buf(const char *buf, int len)
         return;
 
     uart_irq_tx_enable(dev_upload);
-    data_transmitted = false;
-    int bytes = 0;
-    while (bytes != len) {
-        buf += bytes;
-        len -= bytes;
-        bytes = uart_fifo_fill(dev_upload, (const u8_t *)buf, len);
-    }
-    while (data_transmitted == false) {
-        ;
+    int sent = 0;
+
+    while (len > 0) {
+        data_transmitted = false;
+        sent = uart_fifo_fill(dev_upload, (const u8_t *)buf, len);
```

Comment:
The docs for uart_fifo_fill seem to indicate this should only be called from an ISR, but we're calling this function all over the place I think. Maybe we need to go back to UART samples and see how it's properly used. Or the docs could be wrong; haven't looked into it myself.